spanner/api/Program.cs

```diff
@@ -0,0 +1,989 @@
+                using (var reader = await cmd.ExecuteReaderAsync())
+                {
+                    while (await reader.ReadAsync())
+                    {
+                        Console.WriteLine($"SingerId : {reader["SingerId"]} "
+                        + $"AlbumId : {reader["AlbumId"]} "
+                        + $"AlbumTitle : {reader["AlbumTitle"]}");
+                    }
+                }
+            }
+            return;
+            // [END query_data]
+        }
+
+
+        public static async Task QueryDataWithIndexAsync(
+            string projectId, string instanceId, string databaseId,
+            string startTitle = "Aardvark", string endTitle = "Goo")
+        {
+            Console.WriteLine("Running QueryDataWithIndex...");
+            // [START query_data_with_index]
+            // [START read_data_with_index]
+            string connectionString =
+            $"Data Source=projects/{projectId}/instances/{instanceId}"
+            + $"/databases/{databaseId}";
+            // Create connection to Cloud Spanner.
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                var cmd = connection.CreateSelectCommand(
+                    "SELECT AlbumId, AlbumTitle, MarketingBudget FROM Albums@ "
+                    + "{FORCE_INDEX=AlbumsByAlbumTitle} "
+                    + $"WHERE AlbumTitle >= '{startTitle}' "
+                    + $"AND AlbumTitle < '{endTitle}'");
+                using (var reader = await cmd.ExecuteReaderAsync())
+                {
+                    while (await reader.ReadAsync())
+                    {
+                        Console.WriteLine($"AlbumId : {reader["AlbumId"]} "
+                        + $"AlbumTitle : {reader["AlbumTitle"]} "
+                        + $"MarketingBudget : {reader["MarketingBudget"]}");
+                    }
+                }
+            }
+            return;
+            // [END read_data_with_index]
+            // [END query_data_with_index]
+        }
+
+        public static async Task QueryDataWithStoringIndexAsync(
+            string projectId, string instanceId, string databaseId,
+            string startTitle = "Aardvark", string endTitle = "Goo")
+        {
+            // [START read_data_with_storing_index]
+            string connectionString =
+            $"Data Source=projects/{projectId}/instances/{instanceId}"
+            + $"/databases/{databaseId}";
+            // Create connection to Cloud Spanner.
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                var cmd = connection.CreateSelectCommand(
+                    "SELECT AlbumId, AlbumTitle, MarketingBudget FROM Albums@ "
+                    + "{FORCE_INDEX=AlbumsByAlbumTitle2} "
+                    + $"WHERE AlbumTitle >= '{startTitle}' "
+                    + $"AND AlbumTitle < '{endTitle}'");

```

Comment:
do you also have a sample somewhere that shows parameters used in the select statement as opposed to inline values?